docs/advanced/flash.md

```diff
@@ -118,3 +122,13 @@ They test all flash API functionality. To run the tests, use these commands:
 
 - FlashIAP: `mbed test -m TARGET_NAME -n tests-mbed_drivers-flashiap`.
 - Flash HAL: `mbed test -m TARGET_NAME -n tests-mbed_hal-flash`.
+
+## Troubleshooting
```

Comment:
@ashok-rao I added Troubleshooting here to summarize what could go wrong. If there's anything that bootloader should also pinpoint, let me know, can be added.